code/modules/games/cards/playing_cards.dm

```diff
@@ -24,6 +24,10 @@
 		hand.currenthand.Remove(card)
 		user.put_in_hands(card)
 		hand.update_icon()
+		if(hand.currenthand.len == 1)
+			var/obj/item/toy/singlecard/C = hand.currenthand[1]
+			qdel(hand)
```

Comment:
`C` is what is being placed into the hand, not `hand`. I don't see how `hand` could ever be deleted before `C` is created and defined.